src/translations.cpp

```diff
@@ -73,14 +73,19 @@ std::string locale_dir()
     return loc_dir;
 }
 
-void set_language()
+//if called without any parameters sets the language to the one selected in the options menu
```

Comment:
This makes the function to serve multiple responsibilities at once:

- Get language option
- Set to a specific language

And you made this function to perform its different responsibilities depending on a default parameter.

It would be better if you could break them to their respective functions for better readability.

Reply:
Earlier today I had it separated into two functions: "set_language()" currently present in the game and "toggle_language()" which essentially copied the body of the first function and did almost the same, but duplicated the size.
It looked like this:  https://pastebin.com/aW4k21Ah
Should I go back to that?

Reply:
Of course duplicating code is bad. I mean separating the responsibilities. Currently you squeezed two different tasks in one function:

- Get language settings
- Set the game to a specific language

Reply:
To be completely honest, I'm not entirely sure I'm following ^_^' the reason is that "set_language()" is currently part of the game and is doing those two things as well, is getting the language set in the option and is setting the language to it.
So I guess I should start splitting that into two before adding mine? I will try  :  )

Reply:
It made sense before because it would always get language option and set to that language. But now you've introduced a defaulted parameter and depending on whether the parameter is provided or not, the function can either

- Get language option and set to that language
- Set language to the parameter provided

I don't feel it's still a clean and readable interface design.

Reply:
I think I got you now, sorry if I am a bit slow to see these things, maybe it will come with time ^_^' 
I think I've made the correct changes, I'll compile and then then push it in 10 minutes